apps/src/redux/layout.ts

```diff
@@ -0,0 +1,31 @@
+/** @file Track the app's layout */
+import {PayloadAction, createSelector, createSlice} from '@reduxjs/toolkit';
+
+interface LayoutState {
+  visualizationScale: number | null;
+}
+
+const initialState: LayoutState = {
+  visualizationScale: null,
+};
+
+const layoutSlice = createSlice({
+  name: 'layout',
+  initialState,
+  reducers: {
+    setVisualizationScale(state, action: PayloadAction<number>) {
+      state.visualizationScale = action.payload;
+    },
+  },
+});
+
+export const {setVisualizationScale} = layoutSlice.actions;
+
+const selectSelf = (state: LayoutState) => state;
```

Comment:
should `selectSelf` return `state.layout`? That's what `getRoot` returned previously.

Reply:
I followed [this guide](https://redux-toolkit.js.org/api/createSelector) to learn more about selectors (what getters are now called in redux toolkit).

In the brave new typescript world, `state.layout` doesn't exist because `LayoutState` only has the `visualizationScale` property so I think it's correct to do `state => state => state?.visualizationScale` instead of `state => state.layout => state.layout.visualizationScale`

Reply:
[MakerStatusOverlay](https://github.com/code-dot-org/code-dot-org/blob/67f93ee70f88e5bbe27831a2163d680c5bd4c460/apps/src/lib/kits/maker/ui/MakerStatusOverlay.jsx#L78) is the only place I could find where we are using this function, and we are passing in the entire state (which is standard for a selector). To make is work with a typescript type you can type it like:
```
const selectSelf = (state: {layout: LayoutState}) => state.layout;
```
I did something similar in [this PR](https://github.com/code-dot-org/code-dot-org/pull/51241/files#diff-a1544d57bb69a905a6893b778c8d3861864b663015c586ad5bf03e94af7590caR265).